### PR TITLE
Breeding documentation

### PR DESCRIPTION
**Pull request by SafwatHalaby**

### CODE REVIEW

File: src/Mobs/PassiveMonster.h [hunk 1]
```diff
@@ -27,17 +27,32 @@ class cPassiveMonster :
...
 	bool IsInLove() const { return (m_LoveTimer > 0); }
+
+	/** Returns whether the monster is tired of breeding and is in the cooldown state. */
 	bool IsInLoveCooldown() const { return (m_LoveCooldown > 0); }
 
 protected:
+	/** The monster's breeding partner. */
 	cPassiveMonster * m_LovePartner;
+
+	/** If above 0, the monster is in love mode, and will breed if a nearby mode is also in love mode. Decrements by 1 per tick till reaching zero. */
```
Comment (madmaxoft) [inline src/Mobs/PassiveMonster.h]: `nearby mode`?

Comment (SafwatHalaby) [inline src/Mobs/PassiveMonster.h]: Sorry. Nearby monster.

File: src/Mobs/PassiveMonster.cpp [hunk 2]
```diff
@@ -133,7 +138,7 @@ void cPassiveMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 				virtual bool Item(cEntity * a_Entity) override
 				{
-					// if we're the same species as someone around and they dont have a partner, swipe right
+					// if we're the same species as someone around and they dont have a partner, swipe right.
```
Comment (SafwatHalaby) [inline src/Mobs/PassiveMonster.cpp]: don't

**Comment by SafwatHalaby** [comment]
Squashed. lgtm.

File: src/Mobs/PassiveMonster.cpp [hunk 3]
```diff
@@ -172,7 +177,7 @@ void cPassiveMonster::OnRightClicked(cPlayer & a_Player)
 {
 	super::OnRightClicked(a_Player);
 
-	// if right clicked on the player with breeding items, go into lovemode
+	// if right clicked on the player with breeding items, go into lovemode.
```
Comment (madmaxoft) [inline src/Mobs/PassiveMonster.cpp]: `If a player holding breeding items right-clicked me, go into love mode`

File: src/Mobs/PassiveMonster.cpp [hunk 4]
```diff
@@ -133,7 +138,7 @@ void cPassiveMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 				virtual bool Item(cEntity * a_Entity) override
 				{
-					// if we're the same species as someone around and they dont have a partner, swipe right
+					// if we're the same species as someone around and they don't have a partner, swipe right.
```
Comment (madmaxoft) [inline src/Mobs/PassiveMonster.cpp]: `swipe right`?

File: src/Mobs/PassiveMonster.cpp (hunk 3 above)
Comment (SafwatHalaby) [inline src/Mobs/PassiveMonster.cpp]: Fixed.

File: src/Mobs/PassiveMonster.cpp (hunk 4 above)
Comment (SafwatHalaby) [inline src/Mobs/PassiveMonster.cpp]: [Urban dictionary](http://he.urbandictionary.com/define.php?term=Swipe+right): 

> A phrase used to describe your acceptance of something. The term was originally a reference to the Tinder app. On Tinder, swiping right means you approve of a male/female after judging them by a few picture and a short bio. 
> Example usage: That girl Hannah over there is really hott. I would definitely swipe right on her.

Comment (madmaxoft) [inline src/Mobs/PassiveMonster.cpp]: If I need to go visit urban dictionary to understand something, then it's not a good comment.

Comment (SafwatHalaby) [inline src/Mobs/PassiveMonster.cpp]: I have a feeling @tigerw will come up with a joke and post it here. Anyways, I'll change it for something less urban.

**Comment by SphinxC0re** [comment]
Could you change the commit message to something that explains what you changed? Like
```Added documentation for breeding code```

File: src/Mobs/PassiveMonster.cpp (hunk 4 above)
Comment (SafwatHalaby) [inline src/Mobs/PassiveMonster.cpp]: >If I need to go visit urban dictionary to understand something, then it's not a good comment.

I agree. This was not introduced by me.

**Comment by SphinxC0re** [comment]
Also could you please work on your own branch inside the repo and not in your fork? Makes collaborating very difficult

**Comment by SafwatHalaby** [comment]
@SphinxC0re Alright, I will do so in the future.

**Comment by SafwatHalaby** [comment]
I rebased and renamed the commit.

**Comment by SafwatHalaby** [comment]
What is the logic in working at the main repo? Wouldn't it be chaoatic for two people to work on a single branch?

**Comment by SphinxC0re** [comment]
No everyone is working on their own branch. I'm just saying it would be easier if two or more ppl are working on the same project. For you it would be easier to update because you just do ```git pull```

**Comment by madmaxoft** [comment]
It is possible for two people to "work" on the same branch, in the sense of "I see a typo, I fix it"

**Comment by SphinxC0re** [comment]
Yup :+1:

**Comment by SafwatHalaby** [comment]
I removed the extra dots. lgtm.

**Comment by SphinxC0re** [comment]
No, you didn't :smile: there are still dots in the second file you've edited

**Comment by SafwatHalaby** [comment]
But all doxy-comments seem to have those.

**Comment by SafwatHalaby** [comment]
Actually, not all. The codebase is not consistent with regards to dots in doxy comments. Should I remove them?

File: src/Mobs/PassiveMonster.cpp (hunk 4 above)
Comment (tigerw) [inline src/Mobs/PassiveMonster.cpp]: I, ah, have something in mind, but maybe not.

Instead: what sort of tinder?
Tinder

**Comment by SphinxC0re** [comment]
That would be good. It would also be nice to harden the comment specification on the CONTRBUTING file

**Comment by SphinxC0re** [comment]
@madmaxoft The dots haven't been removed

**Comment by madmaxoft** [comment]
There are no dots in the code comments, and I don't care that much about dots in doxy comments - they are complete sentences after all, unlike the in-code comments. I'm more concerned about the line lengths in those doxy comments, they could have used some wrapping, but whatever.

**Pull request merged**